### PR TITLE
541C - Cleanup unused functions/classes given CarbonAwareParameters

### PR DESCRIPTION
**Pull request by JenMadiedo**
Issue Number: 541

## Summary
Cleanup of all unused functions/classes given the switch to CarbonAwareParameters. 

## Changes

- Remove `Constants.cs`
- Remove `CarbonAwareConstants.cs`
- Remove all functions of the form `GetXxOrThrow()` in aggregator (all taken care of by CarbonAwareParameters class now)
- Remove `GetAndValidateForecastInput()` function (take care of by CarbonAwareParameters class now)
- Remove unused dependencies in .csproj files

## Checklist

- [X] Local Tests Passing?
- [X] CICD and Pipeline Tests Passing?
- [ ] Added any new Tests?
- [ ] Documentation Updates Made?
- [ ] Are there any API Changes? If yes, please describe below.
- [X] This is not a breaking change. If it is, please describe it below.

## Are there API Changes?
N/A

## Is this a breaking change?
N/A

## Anything else?
Other comments, collaborators, etc.
This PR Closes Issue #

### CODE REVIEW

File: src/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareParameters.cs [hunk 1]
```diff
@@ -182,6 +182,18 @@ public DateTimeOffset GetEndOrDefault(DateTimeOffset defaultValue)
         return parameters;
     }
 
+    /// <summary>
+    /// Get an IEnumerable of all the PropertyNames defined in the CarbonAwareParameters class
+    /// </summary>
+    public static IEnumerable<PropertyName> GetPropertyNames() {
```
Comment (JenMadiedo) [inline src/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareParameters.cs]: Made public so can be used by tests

File: src/CarbonAware.Aggregators/test/CarbonAwareParametersTests.cs [hunk 2]
```diff
@@ -1,6 +1,5 @@
 using CarbonAware.Aggregators.CarbonAware;
 using CarbonAware.Model;
-using DevLab.JmesPath.Functions;
```
Comment (bderusha) [inline src/CarbonAware.Aggregators/test/CarbonAwareParametersTests.cs:3]: 😕

File: src/CarbonAware.Aggregators/test/CarbonAwareParametersTests.cs [hunk 3]
```diff
@@ -219,7 +217,7 @@ public void PropertyNameEnum_MatchesModelAndDtoClassProperties()
         // Arrange
         var dtoClassPropertyNames = typeof(CarbonAwareParametersBaseDTO).GetProperties().Select(p => p.Name);
         var modelClassPropertyNames = typeof(CarbonAwareParameters).GetProperties().Select(p => p.Name);
-        var enumPropertyNames = Enum.GetNames<PropertyName>();
+        var enumPropertyNames = CarbonAwareParameters.GetPropertyNames();
```
Comment (bderusha) [inline src/CarbonAware.Aggregators/test/CarbonAwareParametersTests.cs]: Does this test pass this way even though `enumPropertyNames` is an `IEnumerable<PropertyName>` type and the other two vars are `IEnumerable<string>`types?

File: src/CarbonAware.Tools/CarbonAware.Tools.WattTimeClient/src/CarbonAware.Tools.WattTimeClient.csproj [hunk 4]
```diff
@@ -16,13 +16,7 @@
...
   <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.Configuration" Version="6.0.1" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="6.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="6.0.0" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="6.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="6.0.0" />
-    <PackageReference Include="Microsoft.Net.Http.Headers" Version="2.2.8" />
-    <PackageReference Include="WireMock.Net" Version="1.4.43" />
+	<PackageReference Include="Microsoft.Net.Http.Headers" Version="2.2.8" />
```
Comment (bderusha) [inline src/CarbonAware.Tools/CarbonAware.Tools.WattTimeClient/src/CarbonAware.Tools.WattTimeClient.csproj]:
```suggestion
    <PackageReference Include="Microsoft.Net.Http.Headers" Version="2.2.8" />
```

File: src/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareParameters.cs [hunk 5]
```diff
@@ -182,6 +182,18 @@ public DateTimeOffset GetEndOrDefault(DateTimeOffset defaultValue)
...
+        var properties = new List<PropertyName>();
+        foreach (string name in Enum.GetNames<PropertyName>())
```
Comment (juzuluag) [inline src/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareParameters.cs]: A one-liner might accomplish the conversion.

```suggestion
        return Enum.GetNames(typeof(PropertyName)).ToList();
```

**Review by juzuluag — COMMENTED** [review]
One suggestion. Otherwise lgtm 👍

**Review by juzuluag — APPROVED** [review]
👍

**Review by gfmatthews — APPROVED** [review]
LGTM - no real issues of substance. Thanks for breaking this into digestible PRs.

**Pull request merged**